### PR TITLE
[EDA] People & Groups: Contact Information - Page and Contact Language

### PR DESCRIPTION
**Pull request by m-kinkade**
# Critical Changes

# Changes

# Issues Closed

# New Metadata
### Apex Classes
ContactLanguageSettingsModel
ContactLanguageSettingsModel_TEST
ContactLanguageSettingsMapper
ContactLanguageSettingsMapper_TEST
ContactLanguageSettingsService
ContactLanguageSettingsService_TEST
ContactInformationSettingsVMapper
ContactInformationSettingsVMapper_TEST
ContactInformationSettingsController
ContactInformationSettingsController_TEST
ContactLanguageSettingsVModel
ContactLanguageSettingsVModel_TEST

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: force-app/main/default/classes/ComboboxVModel.cls [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group Health Check
+ * @group-content ../../ApexDocContent/EdaSettings.htm
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/ComboboxVModel.cls:31]: Let's clear the groups on these truly generic components.
```suggestion
 * @group
 * @group-content 
```

File: force-app/main/default/classes/ComboboxValueVModel.cls [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group Health Check
+ * @group-content ../../ApexDocContent/EdaSettings.htm
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/ComboboxValueVModel.cls:31]: Let's clear the groups on these truly generic components.
```suggestion
 * @group
 * @group-content
```

File: force-app/main/default/classes/ContactInformationSettingsVMapper.cls [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     * in Hierarchy Settings and all available active options.
+     * @return A ContactLanguageSettingsVModel.
+     **********************************************************************************************/
+    public virtual ContactLanguageSettingsVModel getContactLanguageSettingsVModel() {
+        ContactLanguageSettingsModel contactLanguageSettingsModel = this.locateContactLanguageSettingsService()
+            .getContactLanguageSettingsModel();
+
+        Map<String, PicklistEntryModel> activeContactLanguageFluencyPicklistEntryModelsByName = this.locatePicklistEntryService()
+            .getActivePicklistEntryModelsByName(
+                this.locatePicklistEntryService()
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/ContactInformationSettingsVMapper.cls:74]: I'm going to call out that not using a local variable makes it easier to refactor code but we are trading off on detecting which method is throwing an exception if we get to debugging and encounter a problem.

File: force-app/main/default/classes/ContactInfoSettingsController_TEST.cls [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group
+ * @group-content ../../ApexDocContent/Relationships
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/ContactInfoSettingsController_TEST.cls:34]:
```suggestion
 * @group-content
```

File: force-app/main/default/classes/ContactLanguageSettingsVModel_TEST.cls [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group
+ * @group-content ../../ApexDocContent/Relationships
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/ContactLanguageSettingsVModel_TEST.cls:34]:
```suggestion
 * @group-content
```

File: force-app/main/default/lwc/contactInformationSettings/contactInformationSettings.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        return undefined;
+    }
+
+    @wire(getContactLanguageSettingsVModel)
+    contactLanguageSettingsVModel({ error, data }) {
+        if (data) {
+            this.contactLanguageSettingsVModel = data;
+        } else if (error) {
+            console.log("error retrieving contactLanguageSettingsVModel");
```
Comment (MatthewBlanski) [inline force-app/main/default/lwc/contactInformationSettings/contactInformationSettings.js:40]:
```suggestion
            //console.log("error retrieving contactLanguageSettingsVModel");
```

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]
There are a few minor places to perform edits on the groups and group content in the documentation. Also note that relying on a non-dependent branch like this can be a bit dangerous as you're incorporating changes you may not wish to. Once we get this far out it gets a bit dicey. We should try to avoid this in the future.

**Review by MatthewBlanski — APPROVED** [review]

**Comment by spinkelman** [comment]
QA Passes - Merging - Closing WI

**Pull request merged**